src/wui/minimap.cc

```diff
@@ -205,21 +212,25 @@ void MiniMap::toggle(MiniMapLayer const button) {
 }
 
 void MiniMap::resize() {
-	view_.set_zoom((*view_.minimap_layers_ & MiniMapLayer::Zoom2) ? 2 : 1);
+	view_.set_zoom((*view_.minimap_layers_ & MiniMapLayer::Zoom2) ? true : false);
```

Comment:
Please, just use `view_.set_zoom(*view_.minimap_layers_ & MiniMapLayer::Zoom2);` here... o_O

Reply:
Thank you for your review. I addressed some of your comments.


> Overall testing worked good, nice to have a usable minimap on small maps.
> Some observations:
> 
>     * The "red box" that should display the current viewport is too small, no idea what it is showing. This is also visible in your demo-screenshots, e.g., the first one: The red box is a small rectangle in the middle of the player's territory but the viewport even shows the borders. Please fix this one, my further observations are "nice to have".
> 

I'll dig into this.

>     * The zoom button is still disabled after switching to fullscreen. Also, the map might be "lost" outside the screen when switching back to window mode, but this is a problem of all dialogs and most likely out of scope of this branch.
> 
>     * The "is zoomed" state is stored and restored when re-opening the window, even when the resolution of the game window is now too small to contain the zoomed minimap.
> 

That's why I don't disable the zoom button when the map is zoomed. So, the player can still resize the  map. Afterwards, the zoom button gets disabled. Haven't really looked into alternatives though.

>     * Just a thought: The minimap window might be too large for some players, in case they prefer to keep it always open in a corner. But we can deal with that if anyone complains.
> 

I think so, too.

>     * For small map, the minimap does look ugly/pixelated. But well, so be it. Personally, I like it, but no idea what younger players think of it. ^^
> 

I like it as well. But it is the way the map gets drawn that makes it pixelated.

>     * An old bug, already in trunk: After switching to fullscreeen mode, the red box is only recalculated when the map is moved (even though the game window now displays more of the game map)
> 

I'll look into it. 

> The code seems to be okay, though I haven't understood everything and was too lazy to dig deeper. But it is UI code and the result looks good, so I guess it is alright.
> You have a lot of `?:` clauses in the code, I think all of them can be replaced by passing the condition directly as parameter to `scale_map()`. Or is there a reason that you still have them in the code?

No, just me having not slept enough 😄. It is fixed.